web-components/src/components/inputs/validation.ts

```diff
@@ -78,3 +79,18 @@ export function isValidAddress(value: string, prefix?: string): boolean {
     return false;
   }
 }
+
+export function validatePrice(
+  price?: number,
+  maximumFractionDigits?: number,
+): string | undefined {
+  if (!price) {
+    return requiredMessage;
+  }
+  const priceDecimalPlaces = price?.toString().split(/\.|,/)?.[1]?.length;
+  maximumFractionDigits = maximumFractionDigits || MAX_FRACTION_DIGITS;
+  if (!!priceDecimalPlaces && priceDecimalPlaces > maximumFractionDigits) {
+    return `Maximum ${maximumFractionDigits} decimal places`;
+  }
```

Comment:
Added validation here to limit to six decimal places. This avoids a ledger error when the udenom has decimal places.